comp/core/secrets/secretsimpl/secrets_test.go

```diff
@@ -739,3 +740,139 @@ func TestRefreshAllowlistAppliesToEachSettingPath(t *testing.T) {
+			},
+		},
+	}
+	resolver.auditFilename = tmpfile.Name()
+	resolver.auditFileMaxSize = 1000 // enough to add a few rows
+
+	resolver.fetchHookFunc = func(secrets []string) (map[string]string, error) {
+		return map[string]string{
+			"handle": "second_value",
+		}, nil
+	}
+
+	// Refresh the secrets, which will add to the audit file
+	_, err = resolver.Refresh()
+	require.NoError(t, err)
+	assert.Equal(t, auditFileNumRows(tmpfile.Name()), 1)
+
+	resolver.fetchHookFunc = func(secrets []string) (map[string]string, error) {
+		return map[string]string{
+			"handle": "third_value",
+		}, nil
+	}
+
+	// Refresh secrets again, which will add another row the audit file
+	_, err = resolver.Refresh()
+	require.NoError(t, err)
+	assert.Equal(t, auditFileNumRows(tmpfile.Name()), 2)
+
+	resolver.fetchHookFunc = func(secrets []string) (map[string]string, error) {
+		return map[string]string{
+			"handle": "fourth_value",
+		}, nil
+	}
+}
+
+// helper to read number of rows in the audit file
+func auditFileNumRows(filename string) int {
+	data, _ := os.ReadFile(filename)
+	return len(strings.Split(strings.Trim(string(data), "\n"), "\n"))
+}
+
+func TestIsLikelyAPIOrAppKey(t *testing.T) {
+	type testCase struct {
+		name        string
+		handle      string
+		secretValue string
+		origin      handleToContext
+		expect      bool
+	}
+
+	currentTest := t
+	testCases := []testCase{
+		{
+			name:        "looks like an api_key",
+			handle:      "vault://userToken",
+			secretValue: "0123456789abcdef0123456789abcdef",
+			origin: map[string][]secretContext{
+				"vault://userToken": {
+					{
+						origin: "conf.yaml",
+						path:   []string{"provider", "credentials", "apiKey"},
+					},
+				},
+			},
+			expect: true,
+		},
+		{
+			name:        "wrong length",
+			handle:      "vault://userToken",
+			secretValue: "0123456789abcdef0123456789abc",
+			origin: map[string][]secretContext{
+				"vault://userToken": {
+					{
+						origin: "conf.yaml",
+						path:   []string{"provider", "credentials", "apiKey"},
+					},
+				},
+			},
+			expect: false,
+		},
+		{
+			name:        "not hex",
+			handle:      "vault://userToken",
+			secretValue: "0123456789stuvwx0123456789stuvwx",
+			origin: map[string][]secretContext{
+				"vault://userToken": {
+					{
+						origin: "conf.yaml",
+						path:   []string{"provider", "credentials", "apiKey"},
+					},
+				},
+			},
+			expect: false,
+		},
+		{
+			name:        "likely password",
+			handle:      "vault://secretPassword",
+			secretValue: "0123456789abcdef0123456789abcdef",
+			origin: map[string][]secretContext{
+				"vault://userToken": {
+					{
+						origin: "conf.yaml",
+						path:   []string{"provider", "credentials", "password"},
+					},
+				},
+			},
+			expect: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			currentTest = t
+			result := isLikelyAPIOrAppKey(tc.handle, tc.secretValue, tc.origin)
+			assert.Equal(currentTest, tc.expect, result)
+		})
```

Comment:
Just out of curiosity, what is the purpose of the `currentTest` variable ?
Can’t we use directly `t` ?

```suggestion
		t.Run(tc.name, func(t *testing.T) {
			result := isLikelyAPIOrAppKey(tc.handle, tc.secretValue, tc.origin)
			assert.Equal(t, tc.expect, result)
		})
```

Reply:
It was a copy and paste from another test, not needed here.